src/components/modals/ManagePlanDerivationGroupsModal.svelte

```diff
@@ -0,0 +1,362 @@
+  import { base } from '$app/paths';
+  import type { ICellRendererParams } from 'ag-grid-community';
+  import Truck from 'bootstrap-icons/icons/truck.svg?component';
+  import { createEventDispatcher } from 'svelte';
+  import {
+    derivationGroupPlanLinkError,
+    derivationGroups,
+    externalSources,
+    selectedPlanDerivationGroupNames,
+  } from '../../stores/external-source';
+  import { plan } from '../../stores/plan';
+  import { plugins } from '../../stores/plugins';
+  import { view } from '../../stores/views';
+  import type { User } from '../../types/app';
+  import type { DataGridColumnDef } from '../../types/data-grid';
+  import type { DerivationGroup, ExternalSourceSlim } from '../../types/external-source';
+  import type { Layer } from '../../types/timeline';
+  import effects from '../../utilities/effects';
+  import { formatDate } from '../../utilities/time';
+  import { isExternalEventLayer } from '../../utilities/timeline';
+  import { showFailureToast } from '../../utilities/toast';
+  import Collapse from '../Collapse.svelte';
+  import Input from '../form/Input.svelte';
+  import CssGrid from '../ui/CssGrid.svelte';
+  import CssGridGutter from '../ui/CssGridGutter.svelte';
+  import DataGrid from '../ui/DataGrid/DataGrid.svelte';
+  import DataGridActions from '../ui/DataGrid/DataGridActions.svelte';
+  import Panel from '../ui/Panel.svelte';
+  import SectionTitle from '../ui/SectionTitle.svelte';
+  import Modal from './Modal.svelte';
+  import ModalContent from './ModalContent.svelte';
+  import ModalFooter from './ModalFooter.svelte';
+  import ModalHeader from './ModalHeader.svelte';
+
+  export let user: User | null;
+
+  type CellRendererParams = {
+    viewDerivationGroup: (derivationGroup: DerivationGroup) => void;
+  };
+  type DerivationGroupCellRendererParams = ICellRendererParams<DerivationGroup> & CellRendererParams;
+
+  const dispatch = createEventDispatcher<{
+    close: void;
+  }>();
+
+  let externalEventLayers: Layer[] | undefined;
+  let dataGrid: DataGrid<DerivationGroup>;
+  let baseColumnDefs: DataGridColumnDef<DerivationGroup>[] = [];
+
+  selectedPlanDerivationGroupNames.subscribe(() => {
+    if (baseColumnDefs.length > 0 && dataGrid) {
+      // no current way to change just a specific cell unless we add something about plan associations to the DG object,
+      //    which we don't seek to do.
+      // this does mean every update to any entry in selectedPlanDerivationGroupIds refreshes the whole column, and flashes it,
+      //    though that isn't particularly a problem and does do good to signal association complete. Also a small delay, which
+      //    buffers button smashing and repeated updates pretty well!
+      dataGrid.refreshCells();
+    }
+  });
+
+  $: externalEventLayers = $view?.definition.plan.timelines
+    .flatMap(timeline => timeline.rows)
+    .flatMap(row => row.layers)
+    .filter(layer => isExternalEventLayer(layer));
+
+  baseColumnDefs = [
```

Comment:
I think the point of defining it this way was mimicking how we have defined column definitions in other tables, though admittedly this one could easily be merged with `columnDefs`, especially any cellRendering (though a later commend did suggest that we refactor that too). In any case, this will be merged with `columnDefs`.

Reply:
See [this commit](https://github.com/NASA-AMMOS/aerie-ui/pull/1396/commits/5756e9fe9c0f49dd6f0454df66f03882053a7f0a).